test/GameEngine/VoteMechanic_Test.java

```diff
@@ -0,0 +1,44 @@
+package GameEngine;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import Character.CharacterFactory.Roles;
+import GameEngine.GameRegistration.Player;
+
+public class VoteMechanic_Test {
+	
+	@Before
+	public void setup() {
+		GameEngine.Registration.registerPlayer("Derek");
+		GameEngine.Registration.registerPlayer("Andy");
+		GameEngine.Registration.registerPlayer("Connie");
+		GameEngine.Registration.registerPlayer("Harris");
+		GameEngine.Registration.registerPlayer("Eleanor");
+		List<Roles> role_list = new ArrayList<Roles>();
+		role_list.add(Roles.Doctor);
+		role_list.add(Roles.Doctor);
+		role_list.add(Roles.Mayor);
```

Comment:
I'm seeing that Mayor's vote count has been set to 3 when revealed. Can you add an assert into VoteMechanic_Test that tests vote counts / lynch result when mayor is revealed and when mayor is not revealed?